webapp/src/components/MessageForm.tsx

```diff
@@ -125,102 +138,156 @@ const MessageForm: FC<MessageFormProps> = ({
-              </Button>
-            )}
+        <Box sx={{ flex: 1, maxWidth: '100%', overflow: 'hidden' }}>
+          <Typography
+            component="h2"
+            maxWidth="100%"
+            overflow="hidden"
+            textOverflow="ellipsis"
+            whiteSpace="nowrap"
+            width="100%"
+          >
+            {messageIdParts.map((part, i) => (
+              <Typography
+                key={part}
+                color={
+                  i === messageIdParts.length - 1
+                    ? 'text.primary'
+                    : 'text.secondary'
+                }
+                component="span"
+              >
+                {part}
+                {i < messageIdParts.length - 1 && '.'}
+              </Typography>
+            ))}
+          </Typography>
+          <Typography color="text.primary">{message.defaultMessage}</Typography>
+        </Box>
+        <Box
+          sx={{
+            display: 'flex',
+            flex: 1,
+            flexDirection: 'column',
+            rowGap: theme.spacing(1),
+          }}
+        >
+          <TextField
+            aria-readonly={state.translationStatus === 'updating'}
+            fullWidth
+            InputLabelProps={{ shrink: true }}
+            InputProps={{ readOnly: state.translationStatus === 'updating' }}
+            label="Translation"
+            maxRows={4}
+            minRows={4}
+            multiline
+            onChange={onChange}
+            sx={{ flexGrow: 1 }}
+            value={state.translationText}
+          />
+          <ButtonGroup
+            aria-label="Translation actions"
+            sx={{ justifyContent: 'flex-end' }}
+          >
             {(state.translationStatus === 'modified' ||
               state.translationStatus === 'error' ||
               state.translationStatus === 'updating') && (
               <Button
-                aria-label="Reset"
                 disabled={state.translationStatus === 'updating'}
                 onClick={onReset}
-                variant="outlined"
+                startIcon={<RestartAlt />}
               >
-                ↺
+                Reset
               </Button>
             )}
-          </Box>
-          {!!message.params.length && (
-            <Box>
-              <Typography>You can use the following parameters:</Typography>
-              <List
-                sx={{
-                  columnGap: theme.spacing(1),
-                  display: 'flex',
-                  flexDirection: 'row',
-                }}
-              >
-                {message.params.map((param) => {
-                  return (
-                    <ListItem
-                      key={param.name}
-                      sx={{ padding: 0, width: 'auto' }}
-                    >
-                      <code>{param.name}</code>
-                    </ListItem>
-                  );
-                })}
-              </List>
-            </Box>
-          )}
```

Comment:
Let's create an issue for implementing a parameters UI.

Reply:
https://github.com/zetkin/lyra/issues/122

Reply:
We created #122